thepracticaldev/dev.to
# Feature: Setup Vault and AppSecrets Wrapper Class

## What type of PR is this? (check all applicable)

## Description
This PR introduces Vault into our application. Vault will allow us to store and manage secrets in a single secure location no matter how many applications we need to support. I added a wrapper class for it, `AppSecrets`. This wrapper not only talks to Vault, but will also fall back on `ApplicationConfig` if Vault is not setup or a key is not found in Vault. This PR simply sets up the framework for using vault to fetch secrets. We still need to figure out our process for setting Vault up on a per-app basis. 

To ensure an app can only access it's own secrets we will be using [Vault policies](https://learn.hashicorp.com/vault/getting-started/policies#overview). Each app will get its own policy and each policy comes with its own token to give the app access to ONLY its secrets. That token is what will be set as `VAULT_TOKEN`

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/9#card-38621281

## QA Instructions, Screenshots, Recordings
Without any setup ensure the application console and UI loads. This feature is going to be optional so we don't want it hindering anyone who doesn't want to use it. 

## Added tests?


![alt_text](https://thedinglehopper.files.wordpress.com/2014/08/oceans-eleven-vault.jpg)
